src/common/cockpitmemory.c

```diff
@@ -40,33 +38,16 @@ int      cockpit_secmem_drain = 0;
  *
  * This is very similar to memset but we take extra measures to
  * prevent the compiler from optimizing it away.
- *
- * See http://www.dwheeler.com/secure-class/Secure-Programs-HOWTO/protect-secrets.html
  */
 
 void
 cockpit_memory_clear (void * data,
                       ssize_t len)
 {
-  volatile char *vp;
-
-  if (!data)
-    return;
-
   if (len < 0)
     len = strlen (data);
 
-  /* Defeats some optimizations */
-  memset (data, 0xAA, len);
-  memset (data, 0xBB, len);
-
-  /* Defeats others */
-  vp = (volatile char *)data;
-  while (len--)
-    {
-      cockpit_secmem_drain |= *vp;
-      *(vp++) = 0xAA;
-    }
+  explicit_bzero (data, len);
```

Comment:
This looks dangerous and wrong. `explicit_bzero()` expects a `size_t` length argument, and is not documented to have any magic value for length autodetection. OTOH this is meant to be called with -1 (as you mention in your commit message). So you probably need something like `len >= 0 ? len : strlen (data)` here?

That may or may not explain the unit test failures.

Reply:
Eh, TGIF! The `strlen()` is right there, I mis-read the diff.